### PR TITLE
Fix add clip with null clip throws NullReferenceException.

### PR DESCRIPTION
**Pull request by bostelk**
This seems like the most appropriate place to throw the exception. An invalid playable argument exception is thrown otherwise.

### CODE REVIEW

File: Assets/SimpleAnimationComponent/SimpleAnimation.cs [hunk 1]
```diff
@@ -71,6 +71,9 @@ public WrapMode wrapMode
 
     public void AddClip(AnimationClip clip, string newName)
     {
+        if (clip == null)
+            throw new System.NullReferenceException("clip");
```
Comment (davidg-unity) [inline Assets/SimpleAnimationComponent/SimpleAnimation.cs]: Indeed, that's the right place for this, but could you replace NullReferenceException (the object you are interacting with is null), by ArgumentNullException (the argument is null)?

Comment (bostelk) [reply Assets/SimpleAnimationComponent/SimpleAnimation.cs]: Ah, oops. Sure.

**Comment by bostelk** [comment]
I think this test case now fails? Is it correct? [AddClip_WithNullClip_Throws_NullReferenceException](https://github.com/Unity-Technologies/SimpleAnimation/blob/23542431505f469b6fc85cd9b091b17c5f1db337/Assets/SimpleAnimationComponent/Tests/PlaymodeTests/ComparativeTests/ClipManagementTests.cs#L49)

**Comment by davidg-unity** [comment]
😞 
Indeed, it will break that test.
I think throwing the correct exception trumps keeping the behaviour the same between both, especially as long as it throws.

changing the last line of the test to:
Assert.Catch<System.SystemException> (() => { animation.AddClip(null, "test"); });
to catch any System Exception, and renaming the test to
AddClip_WithNullClip_Throws 

would do, while keeping the spirit of the test.

It's not optimal, but it's an expedient solution that doesn't require completely rewriting the test or the test framework to catch NullReferenceException _or_ ArgumentNullException.

**Pull request closed without merging**